tests/python/pants_test/pants_run_integration_test.py

```diff
@@ -172,7 +174,30 @@ def _read_log(filename):
 class PantsRunIntegrationTest(unittest.TestCase):
   """A base class useful for integration tests for targets in the same repo."""
 
-  PANTS_SCRIPT_NAME = 'pants'
+  @property
+  def PANTS_SCRIPT_NAME(self) -> str:
+    # NB: We depend on `pants.pex`, rather than `pants`, because we strip the prefix of source
+    # roots, so `src/python/pants` becomes the directory `pants`
+    # (https://github.com/pantsbuild/pants/pull/8185). We cannot have a directory named `pants` and
+    # a file named `pants`. This is the same reason in
+    # https://github.com/pantsbuild/pants/pull/8105 we added `BUILD_ROOT` as an alternate way to
+    # find the build root, because we could not depend on `./pants` for this purpose.
+    #
+    # NB: this is a generated file that gets automatically generated by `./pants` and
+    # `build-support/bin/bootstrap_pants_pex.sh`.
+    if not Path('pants.pex').exists():
+      warn_or_error(
```

Comment:
@stuhood thanks for the recommendation of doing this instead of `SafePantsRunIntegrationTest`!

I can't figure out how to get this deprecation warning to actually print. I think the issue is that this code is only ran via the Pytest subprocess, which squashes std{out,err} except for failures. But even on an integration test failure, I couldn't get this to print the warning. It does at least raise an exception when the version is set to <1.20.0.dev0.

Reply:
I'd probably just remove the deprecation and fallback, or decide we're going to support the fallback until some time in the future - if it doesn't show, it doesn't add any value :)